### PR TITLE
fix yolox to onnx

### PR DESCRIPTION
**Pull request by www516717402**
fix yolox to onnx

### CODE REVIEW

**Comment by ZwwWayne** [comment]
Lint failed, could you first fix the format issue?

File: mmdet/models/dense_heads/yolox_head.py [hunk 1]
```diff
@@ -278,16 +278,19 @@ def get_bboxes(self,
             flatten_bboxes[..., :4] /= flatten_bboxes.new_tensor(
                 scale_factors).unsqueeze(1)
 
-        result_list = []
+        dets = [], labels = []
```
Comment (sergevkim) [inline mmdet/models/dense_heads/yolox_head.py:281]: You need to separate this line into two lines without a comma

Comment (www516717402) [reply mmdet/models/dense_heads/yolox_head.py]: rewrite already.

File: mmdet/models/dense_heads/yolox_head.py [hunk 2]
```diff
@@ -278,16 +278,20 @@ def get_bboxes(self,
             flatten_bboxes[..., :4] /= flatten_bboxes.new_tensor(
                 scale_factors).unsqueeze(1)
 
-        result_list = []
+        dets = []
+        labels = []
         for img_id in range(len(img_metas)):
```
Comment (RunningLeon) [inline mmdet/models/dense_heads/yolox_head.py:283]: Does the exported ONNX of YOLOX support batch inference and dynamic shape features?

**Comment by ZwwWayne** [comment]
Suggest merging the master branch so that you can fully use the CI.

**Comment by RangiLyu** [comment]
We are going to move the onnx export logic out of the get_bboxes function. Would you like to follow the refectoring in #6003 and add an onnx_export function in yolox?

File: mmdet/models/dense_heads/yolox_head.py (hunk 2 above)
Comment (RunningLeon) [reply mmdet/models/dense_heads/yolox_head.py:283]: any feedback here?

**Comment by mathmanu** [comment]
YOLOX is one of the best light weight models for embedded inference, considering the complexity and accuracy.

Now that https://github.com/open-mmlab/mmdetection/pull/6003 is merged, can support for onnx export of YOLOX be added.

**Pull request closed without merging**